### PR TITLE
[mcp] Add get_span_names tool for discovering span names

### PR DESCRIPTION
**Pull request by sajal004004**
### Summary
Implements the `get_span_names` MCP tool that enables users to discover available span names for a service, with optional regex pattern and span kind filtering. This tool is essential for using `search_traces` effectively since span names can be used to filter trace searches.

### Changes
- ✅ Created types file (`get_span_names.go`) in `internal/types/` directory  
- ✅ Created handler file (`get_span_names.go`) in `internal/handlers/` directory  
- ✅ Created comprehensive test file (`get_span_names_test.go`) with 10 test cases  
- ✅ Registered `get_span_names` tool in `server.go` after `get_services`  
- ✅ Updated ADR-002 to document the new tool  

### Key Features
- **Span Name Discovery**: Lists all available span names for a given service
- **Pattern Filtering**: Supports optional regex pattern matching
- **Span Kind Filtering**: Optional filter by span kind (SERVER, CLIENT, etc.)
- **Limit Control**: Configurable limit (default: 100)
- **Consistent Ordering**: Span names are sorted alphabetically

### Testing & Verification
- ✅ All tests pass (10/10 for get_span_names)
- ✅ **98.7% code coverage** (exceeds 95% requirement)
- ✅ Code formatted and follows established patterns
- ✅  I have run lint and test steps successfully

### Test Cases Covered
1. Successful retrieval with sorting
2. Empty service name validation
3. Pattern filtering with regex
4. Span kind filtering
5. Custom limit application
6. Default limit (100)
7. Invalid regex pattern error handling
8. Storage error handling
9. No operations found (empty result)
10. Pattern matches nothing

Part of #7827

### CODE REVIEW

File: cmd/jaeger/internal/extension/jaegermcp/internal/types/get_span_names.go [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+// SPDX-License-Identifier: Apache-2.0
+
+package types
+
+// GetSpanNamesInput defines the input parameters for the get_span_names MCP tool.
+type GetSpanNamesInput struct {
+	// ServiceName filters by service name (required).
+	ServiceName string `json:"service_name" jsonschema:"Filter by service name (required). Use get_services to discover valid names"`
+
+	// Pattern is an optional regex pattern to filter span names (optional).
```
[the remark this answers is not in the document]
Comment (sajal004004) [reply cmd/jaeger/internal/extension/jaegermcp/internal/types/get_span_names.go]: Fixed in bcb8a16c ✅

File: cmd/jaeger/internal/extension/jaegermcp/internal/handlers/get_span_names.go [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+	if err != nil {
+		return nil, types.GetSpanNamesOutput{}, fmt.Errorf("failed to get span names: %w", err)
+	}
+
+	// Filter by pattern if provided
+	var filteredOps []tracestore.Operation
+	if input.Pattern != "" {
+		pattern, err := regexp.Compile(input.Pattern)
+		if err != nil {
+			return nil, types.GetSpanNamesOutput{}, fmt.Errorf("invalid regex pattern: %w", err)
```
[the remark this answers is not in the document]
Comment (sajal004004) [reply cmd/jaeger/internal/extension/jaegermcp/internal/handlers/get_span_names.go]: Fixed in  bcb8a16c ✅

File: cmd/jaeger/internal/extension/jaegermcp/internal/handlers/get_span_names_test.go [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+			}
+		})
+	}
+}
+
+func TestNewGetSpanNamesHandler(t *testing.T) {
+	mock := &mockGetOperationsQueryService{}
+	handler := &getSpanNamesHandler{queryService: mock}
+	assert.NotNil(t, handler)
+}
```
[the remark this answers is not in the document]
Comment (sajal004004) [reply cmd/jaeger/internal/extension/jaegermcp/internal/handlers/get_span_names_test.go:199]: fixed in 23907201✅

File: cmd/jaeger/internal/extension/jaegermcp/internal/handlers/get_span_names_test.go [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+			_, output, err := handler.handle(context.Background(), &mcp.CallToolRequest{}, tt.input)
+
+			if tt.expectedErr != "" {
+				require.Error(t, err)
+				assert.Contains(t, err.Error(), tt.expectedErr)
+				return
+			}
+
+			require.NoError(t, err)
+			assert.Len(t, output.SpanNames, tt.expectedCount)
```
Comment (yurishkuro) [inline cmd/jaeger/internal/extension/jaegermcp/internal/handlers/get_span_names_test.go]: The output struct is easily comparable, just define expected as the same type as tool output and do a single equals assert.

File: cmd/jaeger/internal/extension/jaegermcp/internal/handlers/get_span_names_test.go [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+				},
+			}
+
+			handler := &getSpanNamesHandler{queryService: mock}
+
+			_, output, err := handler.handle(context.Background(), &mcp.CallToolRequest{}, tt.input)
+
+			if tt.expectedErr != "" {
+				require.Error(t, err)
+				assert.Contains(t, err.Error(), tt.expectedErr)
```
Comment (yurishkuro) [inline cmd/jaeger/internal/extension/jaegermcp/internal/handlers/get_span_names_test.go]: require.ErrorContains

File: cmd/jaeger/internal/extension/jaegermcp/internal/types/get_span_names.go [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+	SpanNames []SpanNameInfo `json:"span_names,omitempty" jsonschema:"List of span names for the service"`
+}
+
+// SpanNameInfo contains information about a span name.
+type SpanNameInfo struct {
+	// Name is the span name.
+	Name string `json:"name" jsonschema:"Span name"`
+
+	// SpanKind is the span kind (e.g., SERVER, CLIENT).
+	SpanKind string `json:"span_kind,omitempty" jsonschema:"Span kind if available"`
```
Comment (yurishkuro) [inline cmd/jaeger/internal/extension/jaegermcp/internal/types/get_span_names.go]: Why omitempty? Span kind is optional as input but not as output

File: docs/adr/002-mcp-server.md [hunk 7]
```diff
@@ -77,13 +77,22 @@ tools:
...
     output: List of service names (strings)
 
+  - name: get_span_names
+    description: List available span names for a service. Useful for discovering valid span names before using search_traces.
+    input_schema:
+      service_name: string (required) - Filter by service name. Use get_services to discover valid names.
+      pattern: string (optional) - Optional regex pattern to filter span names
+      span_kind: string (optional) - Optional span kind filter (e.g., SERVER, CLIENT, PRODUCER, CONSUMER, INTERNAL)
+      limit: integer (optional, default: 100) - Maximum number of span names to return
+    output: List of span names with optional span kind information
```
Comment (yurishkuro) [inline docs/adr/002-mcp-server.md]:
```suggestion
    output: List of span names with span kind information
```

File: docs/adr/002-mcp-server.md [hunk 8]
```diff
@@ -77,13 +77,22 @@ tools:
...
+
   - name: search_traces
     description: Find traces matching service, time, attributes, and duration criteria. Returns metadata only.
     input_schema:
-      start_time_min: string (required) - Start of time interval. Supports RFC3339 or relative (e.g., "-1h", "-30m")
+      start_time_min: string (optional, default: "-1h") - Start of time interval. Supports RFC3339 or relative (e.g., "-1h", "-30m")
       start_time_max: string (optional) - End of time interval. Supports RFC3339 or relative (e.g., "now", "-1m"). Default: now
       service_name: string (required) - Filter by service name. Use get_services to discover valid names.
-      operation_name: string (optional) - Filter by operation/span name
+      span_name: string (optional) - Filter by span name. Use get_span_names to discover valid names.
```
[the remark this answers is not in the document]
Comment (yurishkuro) [reply docs/adr/002-mcp-server.md:95]: Span name is the right thing to use in the api

File: cmd/jaeger/internal/extension/jaegermcp/internal/types/get_span_names.go [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+	SpanNames []SpanNameInfo `json:"span_names,omitempty" jsonschema:"List of span names for the service"`
+}
+
+// SpanNameInfo contains information about a span name.
+type SpanNameInfo struct {
+	// Name is the span name.
+	Name string `json:"name" jsonschema:"Span name"`
+
+	// SpanKind is the span kind (e.g., SERVER, CLIENT).
+	SpanKind string `json:"span_kind" jsonschema:"Span kind if available"`
```
Comment (yurishkuro) [inline cmd/jaeger/internal/extension/jaegermcp/internal/types/get_span_names.go]:
```suggestion
	SpanKind string `json:"span_kind" jsonschema:"Span kind"`
```

File: cmd/jaeger/internal/extension/jaegermcp/internal/handlers/get_span_names_test.go (hunk 4 above)
Comment (sajal004004) [reply cmd/jaeger/internal/extension/jaegermcp/internal/handlers/get_span_names_test.go]: done

File: cmd/jaeger/internal/extension/jaegermcp/internal/handlers/get_span_names_test.go (hunk 5 above)
Comment (sajal004004) [reply cmd/jaeger/internal/extension/jaegermcp/internal/handlers/get_span_names_test.go]: done

File: cmd/jaeger/internal/extension/jaegermcp/internal/types/get_span_names.go (hunk 6 above)
Comment (sajal004004) [reply cmd/jaeger/internal/extension/jaegermcp/internal/types/get_span_names.go]: done

File: cmd/jaeger/internal/extension/jaegermcp/internal/types/get_span_names.go [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+	// SpanKind filters by span kind (optional, e.g., SERVER, CLIENT, PRODUCER, CONSUMER, INTERNAL).
+	SpanKind string `json:"span_kind,omitempty" jsonschema:"Optional span kind filter (e.g. SERVER CLIENT PRODUCER CONSUMER INTERNAL)"`
+
+	// Limit is the maximum number of span names to return (optional, default: 100).
+	Limit int `json:"limit,omitempty" jsonschema:"Maximum number of span names to return (default: 100)"`
+}
+
+// GetSpanNamesOutput defines the output of the get_span_names MCP tool.
+type GetSpanNamesOutput struct {
+	SpanNames []SpanNameInfo `json:"span_names,omitempty" jsonschema:"List of span names for the service"`
```
Comment (yurishkuro) [inline cmd/jaeger/internal/extension/jaegermcp/internal/types/get_span_names.go]: * remove omitempty
* add a test that checks that if the result list is empty it is returned as empty array in the JSON `"span_names": []`

Comment (sajal004004) [reply cmd/jaeger/internal/extension/jaegermcp/internal/types/get_span_names.go]: fixed in  commit  "c98d087"
